src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenFunctionPointersTests.cs

```diff
@@ -2306,6 +2325,963 @@ internal unsafe void CM(B b)
+        }
+
+        [Fact]
+        public void TypeArgumentSpecifiedOrInferred()
+        {
+            var verifier = CompileAndVerifyFunctionPointers(@"
+using System;
+unsafe class C
+{
+    static void M1<T>(T t) => Console.Write(t);
+    static void Main()
+    {
+        delegate*<int, void> ptr = &C.M1<int>;
+        ptr(1);
+        ptr = &C.M1;
+        ptr(2);
+    }
+}", expectedOutput: "12");
+
+            verifier.VerifyIL("C.Main()", expectedIL: @"
+{
+  // Code size       29 (0x1d)
+  .maxstack  2
+  .locals init (delegate*<int,void> V_0)
+  IL_0000:  ldftn      ""void C.M1<int>(int)""
+  IL_0006:  stloc.0
+  IL_0007:  ldc.i4.1
+  IL_0008:  ldloc.0
+  IL_0009:  calli      0x3
+  IL_000e:  ldftn      ""void C.M1<int>(int)""
+  IL_0014:  stloc.0
+  IL_0015:  ldc.i4.2
+  IL_0016:  ldloc.0
+  IL_0017:  calli      0x3
+  IL_001c:  ret
+}
+");
+        }
+
+        [Fact]
+        public void ReducedExtensionMethod()
+        {
+            var comp = CreateCompilationWithFunctionPointers(@"
+unsafe static class CHelper
+{
+    public static void M1(this C c) {}
+}
+unsafe class C
+{
+    static void M(C c)
+    {
+        delegate*<C, void> ptr1 = &c.M1;
+        delegate*<void> ptr2 = &c.M1;
+    }
+}");
+
+            comp.VerifyDiagnostics(
+                // (10,35): error CS8757: No overload for 'M1' matches function pointer 'delegate*<C,void>'
+                //         delegate*<C, void> ptr1 = &c.M1;
+                Diagnostic(ErrorCode.ERR_MethFuncPtrMismatch, "&c.M1").WithArguments("M1", "delegate*<C,void>").WithLocation(10, 35),
+                // (11,32): error CS8778: Cannot use a an extension method with a receiver as the target of a '&amp;' operator.
+                //         delegate*<void> ptr2 = &c.M1;
+                Diagnostic(ErrorCode.ERR_CannotUseReducedExtensionMethodInAddressOf, "&c.M1").WithLocation(11, 32)
+            );
+        }
+
+        [Fact]
+        public void UnreducedExtensionMethod()
+        {
+            var verifier = CompileAndVerifyFunctionPointers(@"
+#pragma warning suppress CS0414 // Field never used
+using System;
+unsafe static class CHelper
+{
+    public static void M1(this C c) => Console.Write(c.i);
+}
+unsafe class C
+{
+    public int i;
+    static void Main()
+    {
+        delegate*<C, void> ptr = &CHelper.M1;
+        var c = new C();
+        c.i = 1;
+        ptr(c);
+    }
+}", expectedOutput: "1", options: TestOptions.UnsafeReleaseExe.WithConcurrentBuild(false) /* PROTOTYPE(func-ptr): re-enable on merge */);
+
+            verifier.VerifyIL("C.Main()", expectedIL: @"
+{
+  // Code size       28 (0x1c)
+  .maxstack  3
+  .locals init (C V_0, //c
+                delegate*<C,void> V_1)
+  IL_0000:  ldftn      ""void CHelper.M1(C)""
+  IL_0006:  newobj     ""C..ctor()""
+  IL_000b:  stloc.0
+  IL_000c:  ldloc.0
+  IL_000d:  ldc.i4.1
+  IL_000e:  stfld      ""int C.i""
+  IL_0013:  stloc.1
+  IL_0014:  ldloc.0
+  IL_0015:  ldloc.1
+  IL_0016:  calli      0x4
```

Comment:
>0x4 [](start = 23, length = 3)

It looks like in release build we are getting ```0x5```

Reply:
Yeah, I'm going to merge master into my branch, get Neal to look at this since he last reviewed, and then rebase after sign-off.